### PR TITLE
docs: update bitbucket server setup instructions for v7.20 and above

### PR DESCRIPTION
**Pull request by kopancek**
## Test plan

Verified locally that the docs render OK.

The docs changes [can be seen online here](https://docs.sourcegraph.com/@milan_update_bitbucket_docs/admin/external_service/bitbucket_server#repository-permissions).

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
LGTM! Very clear step-by-step instructions.

Added some comments to comply with the initial **bold** style of button/field names which you began with but then switched to *italics*/`mono`. Feel free to drop some of them, but I think it's better to pick one style and use it consistently (my proposal is **bold**) :)

**Comment by mrnugget** [comment]
Very cool. Can we use some formatting, though, to make it clearer which instructions belong to which screenshot?

Maybe use sub-headings for separate steps, or numbered list for each step?

File: doc/admin/external_service/bitbucket_server.md [hunk 1]
```diff
@@ -54,6 +54,61 @@ This section walks you through the process of setting up an *Application Link be
...
+Click on **Create** link button. 
+
+![Screenshot of Bitbucket Application Links page.](https://storage.googleapis.com/sourcegraph-assets/docs/images/administration/config/external-services/bitbucket_server/application_links_landing_page.png)
+
+Make sure **Atlassian product** is selected (This will probably look confusing, but it is the only way to setup legacy OAuth app that Sourcegraph supports).
+Write Sourcegraph's external URL in the **Application URL** field and click **Continue**. Click **Continue** on the next screen again, even if Bitbucket Server / Bitbucket Data Center warns you about the given URL not responding.
+
+![Screenshot of Bitbucket Create Link form.](https://storage.googleapis.com/sourcegraph-assets/docs/images/administration/config/external-services/bitbucket_server/create_link_part_1.png)
+
+Write `Sourcegraph` as the *Application Name* and select `Generic Application` as the *Application Type*. Leave everything else unset and click **Continue**.
```
[the remark this answers is not in the document]
Comment (kopancek) [reply doc/admin/external_service/bitbucket_server.md:71]: These are not buttons and we previously used the same syntax. So I think we can keep it consistent with what we used before.

File: doc/admin/external_service/bitbucket_server.md [hunk 2]
```diff
@@ -54,6 +54,61 @@ This section walks you through the process of setting up an *Application Link be
...
+
+![Screenshot of second part of Bitbucket Create Link form.](https://storage.googleapis.com/sourcegraph-assets/docs/images/administration/config/external-services/bitbucket_server/create_link_part_2.png)
+
+Now click the edit button in the `Sourcegraph` Application Link that you just created and select the `Incoming Authentication` panel.
+
+![Screenshot of first part of Bitbucket Edit Link form.](https://storage.googleapis.com/sourcegraph-assets/docs/images/administration/config/external-services/bitbucket_server/edit_link_part_1.png)
+
+Generate a *Consumer Key* in your terminal with `echo sourcegraph$(openssl rand -hex 16)`. Copy this command's output and paste it in the *Consumer Key* field. Write `Sourcegraph` in the *Consumer Name* field.
```
[the remark this answers is not in the document]
Comment (kopancek) [reply doc/admin/external_service/bitbucket_server.md:79]: Same as above

File: doc/admin/external_service/bitbucket_server.md [hunk 3]
```diff
@@ -54,6 +54,61 @@ This section walks you through the process of setting up an *Application Link be
...
+
+![Screenshot of second part of Bitbucket Edit Link form.](https://storage.googleapis.com/sourcegraph-assets/docs/images/administration/config/external-services/bitbucket_server/edit_link_part_2.png)
+
+Generate an RSA key pair in your terminal with `openssl genrsa -out sourcegraph.pem 4096 && openssl rsa -in sourcegraph.pem -pubout > sourcegraph.pub`. Copy the contents of `sourcegraph.pub` and paste them in the *Public Key* field.
+
+![Screenshot of third part of Bitbucket Edit Link form.](https://storage.googleapis.com/sourcegraph-assets/docs/images/administration/config/external-services/bitbucket_server/edit_link_part_3.png)
+
+Scroll to the bottom and check the *Allow 2-Legged OAuth* checkbox, then write your admin account's username in the *Execute as* field and, lastly, check the *Allow user impersonation through 2-Legged OAuth* checkbox. Press **Save**.
```
[the remark this answers is not in the document]
Comment (kopancek) [reply doc/admin/external_service/bitbucket_server.md:87]: Same as above

File: doc/admin/external_service/bitbucket_server.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sashaostrikov) [reply doc/admin/external_service/bitbucket_server.md:71]: Well, what I meant are mostly fields, dunno why I wrote "buttons", sorry :)

Here is a screenshot to better express what I meant: you began with `bold` field descriptions and continued with `italics`. I'd choose one type and stick to it
CleanShot 2023-07-28 at 1 33 41

**Comment by kopancek** [comment]
> Very cool. Can we use some formatting, though, to make it clearer which instructions belong to which screenshot?
> 
> Maybe use sub-headings for separate steps, or numbered list for each step?

Used numbered list, but not sure if it's much better. I think we can merge what we have and improve formatting in a followup if we can figure out how. (Mostly trying to make sure to merge everything I can before going on PTO).

**Pull request merged**